event-based-simulators/extras/README.md

```diff
@@ -0,0 +1,104 @@
+Follow the instructions in: https://github.com/SoftwareAG/cumulocity-python-api
+
+### Installation from PyPI
+```shell
+pip install c8y_api
+```
+
+## Run the export script
+If the environment **optional** parameters were not setup, they can be input as arguments when running the script.
+```shell
+ExportProfileData.py [-h]   [--device-ids DEVICE_ID]
+                            [--create-from CREATE_FROM]
+                            [--create-to CREATE_TO]
+                            [--data-type {measurements,alarms,all}]
+                            [--log {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
+                            [--username USERNAME] [--password PASSWORD]
+                            [--baseurl BASEURL] [--tenant-id TENANT-ID]
+
+```
+Example:
+```shell
+python ExportProfileData.py --device-ids 123456 254676 --create-from 2022-11-16T07:39:35.780Z --create-to 2022-11-16T07:49:35.780Z
+```
+
+optional arguments:\
+  -h, --help : show this help message and exit\
+  --device-ids DEVICE_IDS, -i DEVICE_IDS : Input device id / list of device ids\
+  --create-from CREATE_FROM, -from CREATE_FROM : Input "create from" milestone\
+  --create-to CREATE_TO, -to CREATE_TO : Input "create to" milestone\
+  --data-type {measurements,alarms,all}, -d {measurements,alarms,all} : Export "alarms" or "measurements"\
+  --log {DEBUG,INFO,WARNING,ERROR,CRITICAL}, -l {DEBUG,INFO,WARNING,ERROR,CRITICAL} : Log-level\
+  --username USERNAME, -u USERNAME : C8Y Username\
+  --password PASSWORD, -p PASSWORD : C8Y Password\
+  --baseurl BASEURL, -b BASEURL : C8Y Baseurl\
+  --tenant-id TENANT-ID, -t TENANT-ID : C8Y TenantID\
+
+### Credentials Arguments
+Credentials for the C8Y instance can be handed to the script using cli arguments as shown in the example above. The script will try to extract the crendentials from the [Environment File](./Environment.py) if no credentials are presented as arguments.
+
+### Logging
+Log-level: five log levels can be set using the --log argument {DEBUG, INFO, WARNING, ERROR, CRITICAL}. From left to right is the decreasing order of log info amount can be seen: DEBUG>INFO>WARNING>ERROR>CRITICAL. For example, if INFO level is set, DEBUG level messages can not be seen.
+
+### Export time period
+Input both create-from and create-to to set export time. The time format should be: "%Y-%m-%dT%H:%M:%S.%fZ" (i.e 2022-11-14T13:45:15.893Z)
+
+
+## Run the import script
+ 
+```shell
+ImportData.py [-h]  [--ifiles INPUTFILES] 
+                    [--log {DEBUG, INFO, WARNING, ERROR, CRITICAL}] 
+                    [--username USERNAME] [--password PASSWORD]
+                    [--baseurl BASEURL] [--tenant-id TENANT-ID]
+```
+Example:
+```shell
+python ImportData.py --ifiles sim_001_profile sim_002_profile --log DEBUG --username admin --password abcxzy123
+```
+### INPUTFILE 
+Filename (without extension "json") of one or multiple input file can be input. For example: ```sim_001_profile sim_002_profile```.\
```

Comment:
- @pejo-sag @namnhatpham why do we need to specify the filename w/o ".json"; this seems confusing & counter intuitive to me & even auto completion on the command line adds the .json extension
- if I try to import a file (which definitely exists in the export_data folder), it fails
`python ImportData.py -i .\export_data\sim_001`
```
[2022-11-18 10:09:41,962] [INFO] - Connect to tenant t21385998 successfully
--- Logging error ---
Traceback (most recent call last):
  File "C:\_git\oee-simulators\event-based-simulators\extras\ImportData.py", line 123, in LoadFile
    with open(filePath) as f_obj:
FileNotFoundError: [Errno 2] No such file or directory: 'export_data\\.\\export_data\\sim_001.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python310\lib\logging\__init__.py", line 1100, in emit
    msg = self.format(record)
  File "C:\Python310\lib\logging\__init__.py", line 943, in format
    return fmt.format(record)
  File "C:\Python310\lib\logging\__init__.py", line 678, in format
    record.message = record.getMessage()
  File "C:\Python310\lib\logging\__init__.py", line 368, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "C:\_git\oee-simulators\event-based-simulators\extras\ImportData.py", line 177, in <module>
    file_data = LoadFile(filePath)
  File "C:\_git\oee-simulators\event-based-simulators\extras\ImportData.py", line 126, in LoadFile
    fileLogger.error(e, type(e))
Message: FileNotFoundError(2, 'No such file or directory')
Arguments: (<class 'FileNotFoundError'>,)
[2022-11-18 10:09:41,970] [INFO] - Searching for device with ext ID sim_001
[2022-11-18 10:09:42,061] [INFO] - Device(90684536) has been found by its external id "c8y_EventBasedSimulatorProfile/sim_001".
[2022-11-18 10:09:42,061] [INFO] - No Measurements to import
```

Reply:
In my idea, the data files are always in export_data folder and always has json extension so it is unnecessary to make users input them everytime, especially when they need to input multiple filenames. Therefore, I let them to input only the filename without extension json and the script will automatically add the full path and extension.